modules/ComscoreStreamingTag/resources/mw.ComscoreStreamingTag.js

```diff
@@ -0,0 +1,579 @@
+				if (_this.getPlayerPluginState() != _this.PlayerPluginState().SEEKING) {
+					_this.setPlayerPluginState(_this.PlayerPluginState().SEEKING);
+					_this.callGenericPlugin("notify", _this.playerEvents.PAUSE, _this.getLabels(true), _this.getCurrentPosition());
+				}
+			});
+
+			embedPlayer.bindHelper('onOpenFullScreen' + this.bindPostfix, function() {
+				_this.inFullScreen = true;
+				this.genericPlugin.setLabel("ns_st_ws", this.isFullScreen() ? "full" : "norm", true);
+			});
+
+			embedPlayer.bindHelper('onCloseFullScreen' + this.bindPostfix, function() {
+				_this.inFullScreen = false;
+				this.genericPlugin.setLabel("ns_st_ws", this.isFullScreen() ? "full" : "norm", true);
+			});
+
+			embedPlayer.bindHelper( 'onChangeMedia' + _this.bindPostFix, function(){
+				_this.log("onChangeMedia ");
+			});
+
+			embedPlayer.bindHelper( 'onPlayerStateChange' + _this.bindPostFix, function(event){
+				// This code appears to never be called?
+				_this.log("onPlayerStateChange " + event);
+			});
+
+			embedPlayer.bindHelper('onAdOpen' + this.bindPostfix, function(event, adId, networkName, type, index) {
+				_this.currentAd.id = adId;
+				_this.currentAd.type = type;
+				_this.currentAd.index = index;
+
+				// Check if the add is already stored. The ads are played in order.
+				if (_this.adsPlayed.length < index + 1) {
+					_this.adsPlayed.push(_this.currentAd);
+				}
+
+				// We need to update the player plugin state before setting up the clip.
+				_this.setPlayerPluginState(_this.PlayerPluginState().AD_PLAYING);
+				_this.callGenericPlugin("setClip", _this.getClipLabels(), false, {}, true);
+				_this.callGenericPlugin("notify", _this.playerEvents.PLAY, _this.getLabels(), 0);
+
+				_this.shouldSetClip = true;
+			});
+
+			embedPlayer.bindHelper('AdSupport_EndAdPlayback' + this.bindPostfix, function() {
+				_this.setPlayerPluginState(_this.PlayerPluginState().AD_ENDED_PLAYING);
+				_this.callGenericPlugin("notify", _this.playerEvents.END, _this.getLabels());
+				_this.currentAd.id = "";
+				_this.currentAd.type = "";
+				_this.currentAd.index = 0;
+				_this.currentAd.duration = 0;
+			});
+
+			embedPlayer.bindHelper('AdSupport_AdUpdateDuration' + this.bindPostfix, function(event, duration) {
+				_this.currentAd.duration = duration * 1000;
+			});
+
+			embedPlayer.bindHelper('adClick' + this.bindPostfix, function(url) {
+				// When the ad is clicked its also paused
+				_this.callGenericPlugin("notify", _this.playerEvents.PAUSE, _this.getLabels());
+				_this.callGenericPlugin("notify", _this.playerEvents.AD_CLICK, _this.getLabels());
+			});
+
+			// release the player
+			callback();
+		},
+
+		destroy: function() {
+			$( this.embedPlayer ).unbind( this.bindPostfix );
+		},
+
+		getLabels: function(seek) {
+			//get common labels values
+			this.genericPlugin.setLabel("ns_st_br", this.currentBitrate, true);
+			this.genericPlugin.setLabel("ns_st_ws", this.isFullScreen() ? "full" : "norm", true);
```

Comment:
ideally the player would support states for all of this ... you should never have all this extra bind -> set -> store -> get logic for what is just a player state that should have an evaluation mapping. ( This is more for @itaykinnrot & team ( @OrenMe @amirch1  )